backend/app/tests/api/routes/inquiry/test_create_inquiry.py

```diff
@@ -0,0 +1,72 @@
+from unittest.mock import ANY, MagicMock, Mock
+
+import pytest
+from fastapi.testclient import TestClient
+from pytest import MonkeyPatch
+
+import app.services.inquiries
+from app.core.config import settings
+from app.models import Inquiry, InquiryCreate
+
+inquiry_text = "Why do birds suddenly appear every time you are near?"
+
+
+@pytest.fixture(autouse=True)
+def create_inquiry_mock(monkeypatch: MonkeyPatch) -> MagicMock:
+    _mock = MagicMock()
+    monkeypatch.setattr(
+        app.services.inquiries, app.services.inquiries.create_inquiry.__name__, _mock
+    )
+    return _mock
+
+
+@pytest.fixture(autouse=True)
+def get_inquiry_by_text_mock(monkeypatch: MonkeyPatch) -> MagicMock:
+    _mock = MagicMock()
+    monkeypatch.setattr(
+        app.services.inquiries,
+        app.services.inquiries.get_inquiry_by_text.__name__,
+        _mock,
+    )
+    return _mock
+
+
+def test_create_inquiry_api_calls_create_inquiry_service(
+    client: TestClient,
+    superuser_token_headers: dict[str, str],
+    create_inquiry_mock: Mock,
+    get_inquiry_by_text_mock: Mock,
+) -> None:
+    get_inquiry_by_text_mock.return_value = None
+    create_inquiry_mock.return_value = Inquiry(text=inquiry_text)
+    data = {"text": inquiry_text}
+    client.post(
+        f"{settings.API_V1_STR}/inquiries/",
+        headers=superuser_token_headers,
+        json=data,
+    )
+
+    get_inquiry_by_text_mock.assert_called_once()
+    create_inquiry_mock.assert_called_once()
+    create_inquiry_mock.assert_called_with(
+        session=ANY, inquiry_in=InquiryCreate(text=inquiry_text)
+    )
+
+
+def test_create_inquiry_does_not_create_duplicate(
```

Comment:
Not really sure what this is testing from the name.

Reply:
Is it that when it's called with the same inquiry twice? If so, I'm not seeing where that is being done in the test itself...

Reply:
The service layer
(1) Checks if the Inquiry already exists
(2) If it does exist already, it will not create a duplicate.

The mock return value to getInquiry indicates that it does already exist.  Therefore, the createInquiry function is not called

Reply:
```
this is the code being tested

python```
@router.post("/", response_model=InquiryPublic)
def create_inquiry(*, session: SessionDep, inquiry_in: InquiryCreate) -> Inquiry:
    """
    Create new inquiry.
    """
    inquiry = inquiries_service.get_inquiry_by_text(
        session=session, text=inquiry_in.text
    )
    if inquiry:
        raise HTTPException(
            status_code=400,
            detail="This inquiry already exists.",
        )

    return inquiries_service.create_inquiry(session=session, inquiry_in=inquiry_in)
```